ros2_integration/ubuntu/Dockerfile

```diff
@@ -0,0 +1,89 @@
+## Provide Orocos Toolchain (RTT/OCL) on top of the official ROS 2 docker images.
+ARG OROCOS_TOOLCHAIN_BRANCH=ros2
+ARG OROCOS_RTT_ROS_INTEGRATION_BRANCH=master
+ARG ROS_DISTRO
+ARG UBUNTU_DISTRO
+
+## Stage deps: install additional Orocos Toolchain (with CORBA) dependencies
+FROM ros:${ROS_DISTRO}-ros-base-${UBUNTU_DISTRO} AS deps
+RUN apt-get update && apt-get install -y \
+    libboost-all-dev \
+    liblua5.1-0-dev \
+    libncurses5-dev \
+    libnetcdf-dev \
+    libreadline-dev \
+    libxml-xpath-perl \
+    libxml2-dev \
+    omniorb omniidl omniorb-idl omniorb-nameserver libomniorb4-dev \
+    ruby bundler \
+    $([[ "$UBUNTU_DISTRO" != "focal" ]] && echo "ruby-facets") \
+    && rm -rf /var/lib/apt/lists/*
+
+## Stage build: clone and build orocos_toolchain with colcon
+FROM deps AS build
+ARG OROCOS_TOOLCHAIN_BRANCH
+RUN git clone --recursive https://github.com/orocos-toolchain/orocos_toolchain.git -b ${OROCOS_TOOLCHAIN_BRANCH} /build/src/orocos_toolchain
+RUN cd /build \
+    && . /opt/ros/${ROS_DISTRO}/setup.sh \
+    && DESTDIR=/stage colcon build \
+        --event-handlers console_direct+ \
+        --install-base /opt/ros/${ROS_DISTRO} \
+        --merge-install \
+        --cmake-args \
+            \ -DCMAKE_BUILD_TYPE=Release \
+            \ -DCMAKE_FIND_ROOT_PATH=/stage \
+            \ -DCMAKE_PREFIX_PATH=/opt/ros/${ROS_DISTRO} \
+            \ -DENABLE_CORBA=ON \
+            \ -DCORBA_IMPLEMENTATION=OMNIORB \
+            \ -DOROCOS_INSTALL_INTO_PREFIX_ROOT=ON
+
+
+
+## Stage deploy: copy the install-space to /opt/ros/${ROS_DISTRO} as a
+## new image layer based on deps
+FROM deps AS deploy1
+COPY --from=build /stage/opt/ros/${ROS_DISTRO} /opt/ros/${ROS_DISTRO}
+
+FROM deploy1 AS build2
```

Comment:
No need to have a second deployment step. You can just clone all repositories to the same workspace and build them all together in a single `colcon build` invocation, or eventually split in two with `--packages-select orocos_toolchain` and a second invocation with `--packages-skip orocos_toolchain`.

`rtt` is not recognized by `colcon` as a separate package because it finds `orocos_toolchain/CMakeLists.txt` and builds it as a pure CMake package. `rtt` and other toolchain packages are "hidden" within `orocos_toolchain`.

But definitely not two separate build/deploy stages in the same `Dockerfile`. If you have to for whatever reason, please give them more meaningful names, not `build1`, `deploy1`, etc.